lib/cli/interactive-setup/aws-credentials.js

```diff
@@ -86,33 +178,141 @@ const steps = {
       )}\n`
     );
   },
+  // TODO: TESTS
+  handleProviderCreation: async ({ configuration: { org: orgName } }) => {
+    const providersUrl = `https://app.serverless.com/${orgName}/settings/providers?source=cli`;
+    openBrowser(providersUrl);
+    // TODO: CHANGE MESSAGING
+    process.stdout.write('Waiting for creation of provider...\n');
+
+    let onEvent;
+
+    const p = new Promise((resolve) => {
+      let inquirerPrompt;
+
+      const timeoutDuration = 1000 * 60; // 1 minute
+      const showSkipPromptTimeout = setTimeout(() => {
+        inquirerPrompt = inquirer.prompt({
+          message: '\n Press Enter to continue without setting up provider',
+          name: 'skipProviderSetup',
+        });
+
+        inquirerPrompt.then(() => resolve(null));
+      }, timeoutDuration);
+
+      onEvent = (event) => {
+        if (inquirerPrompt) {
+          // Disable inquirer prompt asking to skip without setting provider
+          inquirerPrompt.ui.close();
+        }
+
+        clearTimeout(showSkipPromptTimeout);
+        resolve(event);
+      };
+    });
+
+    // Listen for `provider.created` event to detect creation of new provider
+    const sdk = await getSdkInstance(orgName);
+    await sdk.connect({
+      orgName,
+      onEvent,
+      filter: {
+        events: ['provider.created'],
+      },
+    });
+
+    let maybeEvent;
+    try {
+      maybeEvent = await p;
+    } finally {
+      sdk.disconnect();
+    }
+
+    if (maybeEvent) {
+      process.stdout.write(`\nDetected creation of provider: "${maybeEvent.data.object.alias}"\n`);
```

Comment:
If user created provider, does it also means that it was already assigned to this _app_, or should we issue an additional SDK call (?)

Also I think message facing user should be something like `Assigned provider "<provider-name>"` (in green)

(Wording as _Detected.._ feels as debug log that confirms that internals work, and I don't think we should share such messages)

Reply:
This is where this whole setup method becomes very tricky. The base assumption here is that when a user creates a new provider, it will become a default one so it doesn't have to be explicitly assigned. I was wondering if we should still create an explicit link (see my TODO: comment below) to ensure that it's properly linked after creation (link in the same way as when user selects an existing provider)